### ISSUE TITLE
shell: support additional job environment variables

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#5059]
Came upon some environment variables output by other resource managers that would be convenient if they were also output by Flux.

- job name
- job nodelist
- job timelimit (or in flux parlance, duration)

The most convenient way to get them for a user (I think) is via `flux-jobs`, like `flux jobs -o {name} JOBID`.

I think duration is in the jobspec, `name` could be calculated based on what's the jobspec, and nodelist should be in R, so shouldn't require an additional query of any sort.

**Comment by garlick** [comment]
There is `flux job timeleft` and the C and python timeleft interfaces, which is probably more useful than the requested duration.

Maybe `flux jobs` or the corresponding python is OK for the other two until we find a real use case that can't use it?

In general, I think we're trying not to over-use the environment for communicating such things.

**Comment by chu11** [comment]
> In general, I think we're trying not to over-use the environment for communicating such things.

Makes sense.  I know that some schedulers output like gajillions of variables, but thought a few of the additional common ones might make sense.

**Comment by grondo** [comment]
I'd say not unless there's a specific request or requirement. Once you add a variable you can't take it back. Users can add things like a "name" to the environment in jobspec, and other data above can easily be fetched, or provided via a shell plugin or initrc. 

Maybe close this issue in favor of individual issues for specific environment variables as needed?

**Comment by chu11** [comment]
> Maybe close this issue in favor of individual issues for specific environment variables as needed?

Ok, sounds good.
